### PR TITLE
CHIA-622 Add benchmarks for sha256 operations

### PR DESCRIPTION
**Pull request by jack60612**

### CODE REVIEW

**Comment by jack60612** [comment]
appears to be 10 us (micro seconds faster for test on openssl)

**Review by Rigidity — DISMISSED** [review]

**Review by Rigidity — APPROVED** [review]

**Review by Rigidity — APPROVED** [review]

**Comment by jack60612** [comment]
🥳

File: benches/sha256_hash.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    // setup benchmark
+    let mut group = c.benchmark_group("sha256_hash");
+
+    group.bench_function("hash_benchmark", |b| {
+        b.iter(|| {
+            // this figures out how many iterations to run.
+            for val in 0..MAX_VAL {
+                for len in BYTE_LENGTHS {
+                    let bytes = gen_bytes(val, len);
+                    hash_bytes(&bytes);
```
Comment (arvidn) [inline benches/sha256_hash.rs:32]: there's a chance you need to use `black_box()` here, to prevent the optimizer removing all computation here

Comment (arvidn) [reply benches/sha256_hash.rs:32]: https://docs.rs/criterion/latest/criterion/fn.black_box.html#:~:text=Function%20criterion%3A%3Ablack_box&text=A%20function%20that%20is%20opaque,prevent%20code%20from%20being%20eliminated.

**Pull request merged**